unit_tests/utils/test_datetime_helpers.py

```diff
@@ -262,3 +264,160 @@ def test_epoch_millis():
+    def spy_from_utc(*args, **kwargs):
+        nonlocal whenever_called
+        whenever_called = True
+        # Call the original function
+        return original_instant_module.from_utc(*args, **kwargs)
+
+    # Create a spy for parser.parse
+    def spy_parser_parse(*args, **kwargs):
+        nonlocal dateutil_called
+        dateutil_called = True
+        # Call the original function
+        return original_parser_parse(*args, **kwargs)
+
+    # Create a mock Instant class with our spy methods
+    class MockInstant:
+        @staticmethod
+        def from_timestamp(*args, **kwargs):
+            return spy_from_timestamp(*args, **kwargs)
+
+        @staticmethod
+        def from_utc(*args, **kwargs):
+            return spy_from_utc(*args, **kwargs)
+
+        # Add any other methods that might be called
+        @staticmethod
+        def py_datetime():
+            return original_instant_module.py_datetime()
+
+    # Apply the mocks at the module level
+    monkeypatch.setattr("airbyte_cdk.utils.datetime_helpers.Instant", MockInstant)
+    monkeypatch.setattr("airbyte_cdk.utils.datetime_helpers.parser.parse", spy_parser_parse)
+
+    # Skip formats that would be rejected by validation checks
+    if isinstance(input_value, str) and "March" in input_value:
+        # Skip this test case as it would be rejected by validation
+        return
+
+    # Parse the datetime
+    ab_datetime_parse(input_value)
+
+    # Check which parser was used
+    if expected_parser == "whenever":
+        assert whenever_called, f"Expected whenever parser to be used for {input_value}"
+        assert not dateutil_called, f"Did not expect dateutil parser to be used for {input_value}"
+    else:
+        assert dateutil_called, f"Expected dateutil parser to be used for {input_value}"
+        assert not whenever_called, f"Did not expect whenever parser to be used for {input_value}"
+
+
+def test_whenever_parser_for_iso_formats(monkeypatch):
+    """Test that the whenever parser is used for certain formats even when dateutil is unavailable."""
+
+    # Create a mock dateutil.parser.parse that always raises an exception
+    def mock_parser_parse(dt_str, **kwargs):
+        raise ValueError("dateutil parser is unavailable")
+
+    # Apply the mock at the module level
+    monkeypatch.setattr("airbyte_cdk.utils.datetime_helpers.parser.parse", mock_parser_parse)
+
+    # These formats should still parse correctly using the whenever parser
+    whenever_formats = [
+        "2023-03-14",  # Date-only format
+        1678806566,  # Unix timestamp
+    ]
+
+    for dt_str in whenever_formats:
+        # This should not raise an exception because the whenever parser should be used
+        result = ab_datetime_parse(dt_str)
+        assert isinstance(result, AirbyteDateTime)
+
+
+def test_dateutil_fallback_for_non_iso_formats(monkeypatch):
+    """Test that the dateutil parser is used as a fallback for non-ISO/RFC compliant formats."""
```

Comment:
Ditto. 

The parameterized test seems to cover these cases. Can we delete this test?